src/renderer/hooks/usePaneNodeSearchMenu.tsx

```diff
@@ -0,0 +1,417 @@
+    const matchingNodes = useMemo(
+        () =>
+            getMatchingNodes(searchQuery, schemata.schemata).filter((node) => {
+                if (
+                    !connectingFrom ||
+                    !connectingFromType ||
+                    !connectingFrom.nodeId ||
+                    !connectingFrom.handleId
+                ) {
+                    return true;
+                }
+                if (connectingFrom.handleType === 'source') {
+                    const sourceFn = typeState.functions.get(connectingFrom.nodeId);
+
+                    if (!sourceFn) {
+                        return false;
+                    }
+
+                    const { inOutId } = parseHandle(connectingFrom.handleId);
+                    const sourceType = sourceFn.outputs.get(inOutId);
+
+                    if (!sourceType) {
+                        return false;
+                    }
+
+                    const targetTypes = functionDefinitions.get(node.schemaId);
+
+                    if (!targetTypes) {
+                        return false;
+                    }
+
+                    return [...targetTypes.inputs].some(([number, type]) => {
+                        const overlap = intersect(type, sourceType);
+                        return (
+                            overlap.type !== 'never' &&
+                            schemata.get(node.schemaId).inputs[number].hasHandle
+                        );
+                    });
+                }
+                if (connectingFrom.handleType === 'target') {
+                    const sourceFn = typeState.functions.get(connectingFrom.nodeId);
+
+                    if (!sourceFn) {
+                        return false;
+                    }
+
+                    const { inOutId } = parseHandle(connectingFrom.handleId);
+                    const sourceType = sourceFn.inputs.get(inOutId);
+
+                    if (!sourceType) {
+                        return false;
+                    }
+
+                    const targetTypes = functionDefinitions.get(node.schemaId);
+
+                    if (!targetTypes) {
+                        return false;
+                    }
+
+                    return [...targetTypes.outputDefaults].some(([, type]) => {
+                        const overlap = intersect(type, sourceType);
+                        return overlap.type !== 'never';
+                    });
+                }
+                log.error(`Unknown handle type: ${connectingFrom.handleType!}`);
+                return true;
+            }),
+        [connectingFrom, connectingFromType, searchQuery, schemata.schemata]
+    );
+    const byCategories = useMemo(() => getNodesByCategory(matchingNodes), [matchingNodes]);
+
+    const { favorites } = useNodeFavorites();
+
+    useEffect(() => {
+        setSearchQuery('');
+    }, [connectingFrom]);
+
+    const onPaneContextMenuNodeClick = useCallback(
+        (node: NodeSchema, position: Position) => {
```

Comment:
It's not really a node. Could you rename the parameter or destructure it into `{ schemaId, nodeType }`?